### PR TITLE
Move countrycode to location dimension

### LINKED ISSUE #147: Move countryCode from top-level metadata to an attribute of the location dimension
`countryCode` naturally belongs as an attribute of a `location` dimension.  By specifying `countryCode` this way, we have more flexibility and accuracy in describing a budget dataset that targets multiple countries (for example, [this dataset](https://github.com/openspending/fiscal-data-package-demos/blob/master/eu-allocations-per-ms-2014-2020/datapackage.json#L7) describes allocations for several countries, but the link between the allocation and the country is lost when specified as a top-level array).  Used in combination with `constant`, the behavior of a top-level `countryCode` (i.e. in which all lines refer to the same country) can be retained.

### PR DESCRIPTION
**Pull request by danfowler**
Closes #147

### CODE REVIEW

File: spec/index.md [hunk 1]
```diff
@@ -504,13 +498,25 @@ Whilst strictly not an entity, the concept of an "account" from which money is s
...
+  "dimensionType": "location",
+  "attributes": {
+    "countryCode": {
+      "constant": "GB"
+    }
+  }
+}
+```
 
 Note when applying these as attributes directly on an object we suggest prefixing each value with `geo` so you would have `geoCode` rather than `code` etc.
```
Comment (pwalsh) [inline spec/index.md]: Why?

File: spec/index.md [hunk 2]
```diff
@@ -504,13 +498,25 @@ Whilst strictly not an entity, the concept of an "account" from which money is s
 
 ### Location
 
-There is a frequent desire to label items with location, usually by attaching geographic codes for a region or area. This allows the spending or revenue to  be analysed by region or area. This geographic information can be introduced directly by classifying the item with a code, or, more frequently indirectly by associating a geographic code to e.g. an entity. For example, by labelling a supplier with their location one can then associate a spend with that supplier as spending in that location.
+There is a frequent desire to label items with location, usually by attaching geographic codes for a region or area. This allows the spending or revenue to be analysed by region or area. This geographic information can be introduced directly by classifying the item with a code, or, more frequently indirectly by associating a geographic code to e.g. an entity. For example, by labelling a supplier with their location one can then associate a spend with that supplier as spending in that location.
```
Comment (pwalsh) [inline spec/index.md]: This sentence claims "For example, by labelling a supplier with their location one can then associate a spend with that supplier as spending in that location."

But how does a location dimension "label" a supplier?

File: spec/index.md (hunk 1 above)
Comment (pwalsh) [inline spec/index.md]: Ah, is supposed to mean, when applying location attributes to a *different* dimension?

If yes, it is really hard to grok this: you are actually talking about two use cases:

1. A location dimension
2. Adding location attributes to other dimensions.

File: spec/index.md [hunk 3]
```diff
@@ -504,13 +498,25 @@ Whilst strictly not an entity, the concept of an "account" from which money is s
...
 
 We RECOMMEND using a `location` dimension though attributes may also be applied directly onto another object (e.g. an entity). Here are attributes that `MAY` be applied either directly to an item or to an entity or other object associated to an item.
 
 * `code`: The internal or local geographicCode id based for the geographical region
 * `title`: Name or title of the geographical region targeted by the budget item
-* `codeList`: the geo codelist from which the geocode is drawn 
+* `codeList`: The geo codelist from which the geocode is drawn
```
Comment (pwalsh) [inline spec/index.md]: Could we have an example of how `codeList` is to be applied practically?

File: spec/index.md (hunk 1 above)
Comment (danfowler) [inline spec/index.md]: Re: 2, I would like to simplify this and remove any recommendation to add location attributes to other dimensions: all location information should be within a location dimension.  It is currently more confusing than it needs to be.

Comment (danfowler) [inline spec/index.md]: Re: 2, I would like to simplify this and remove any recommendation to add location attributes to other dimensions: all location information should be within a location dimension.  It is currently more confusing than it needs to be.

Comment (rgrp) [inline spec/index.md]: I'm really confused by this too ...

Comment (danfowler) [inline spec/index.md]: Great, let's remove :smile:

File: spec/index.md (hunk 2 above)
Comment (danfowler) [inline spec/index.md]: This is a reference to a mechanism that I am aiming to remove.

https://github.com/openspending/fiscal-data-package/pull/153/files

**Comment by rufuspollock** [comment]
WONTFIX. Now stale and i think there were issues.

**Pull request closed without merging**